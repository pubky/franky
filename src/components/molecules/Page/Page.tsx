import { ReactNode } from 'react';

import * as Atoms from '@/atoms';
import * as Libs from '@/libs';

interface PageContainerProps {
  children: ReactNode;
  className?: React.HTMLAttributes<HTMLDivElement>['className'];
  as?: 'div' | 'main' | 'section';
  size?: 'default' | 'narrow';
}

interface PageTitleProps extends React.HTMLAttributes<HTMLHeadingElement> {
  className?: React.HTMLAttributes<HTMLHeadingElement>['className'];
  size?: 'medium' | 'large';
}

export function PageContainer({ as: Component = 'div', size = 'default', ...props }: PageContainerProps) {
  const sizeClasses = {
    default: 'container mx-auto max-w-[1200px]',
    narrow: 'container mx-auto max-w-[588px]',
  };

  return (
    <Component data-testid="page-container" className={Libs.cn(sizeClasses[size], props.className)}>
      {props.children}
    </Component>
  );
}

export function PageTitle({ children, className, size = 'large', ...props }: PageTitleProps) {
  const sizeClasses = {
    medium: 'text-4xl lg:text-[60px]',
    large: 'text-5xl lg:text-7xl',
  };

  return (
    <Atoms.Heading {...props} level={1} size="lg" className={Libs.cn(sizeClasses[size], className)}>
      {children}
    </Atoms.Heading>
  );
}
