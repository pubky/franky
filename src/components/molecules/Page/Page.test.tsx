import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { PageContainer, PageTitle } from './Page';

describe('Page Components - Snapshots', () => {
  describe('PageContainer - Snapshots', () => {
    it('matches snapshot for default PageContainer', () => {
      const { container } = render(
        <PageContainer>
          <div>Test content</div>
        </PageContainer>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for PageContainer with different configurations', () => {
      const { container } = render(
        <PageContainer as="main" size="narrow" className="custom-page">
          <p>Main content</p>
        </PageContainer>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('PageTitle - Snapshots', () => {
    it('matches snapshot for default PageTitle', () => {
      const { container } = render(<PageTitle>Default Title</PageTitle>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for PageTitle with different configurations', () => {
      const { container } = render(
        <PageTitle size="medium" className="custom-title">
          Medium Title
        </PageTitle>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
