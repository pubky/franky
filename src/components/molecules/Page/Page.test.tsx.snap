// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page Components - Snapshots > PageContainer - Snapshots > matches snapshot for PageContainer with different configurations 1`] = `
<main
  class="container mx-auto max-w-[588px] custom-page"
  data-testid="page-container"
>
  <p>
    Main content
  </p>
</main>
`;

exports[`Page Components - Snapshots > PageContainer - Snapshots > matches snapshot for default PageContainer 1`] = `
<div
  class="container mx-auto max-w-[1200px]"
  data-testid="page-container"
>
  <div>
    Test content
  </div>
</div>
`;

exports[`Page Components - Snapshots > PageTitle - Snapshots > matches snapshot for PageTitle with different configurations 1`] = `
<h1
  class="font-bold text-foreground text-4xl lg:text-[60px] custom-title"
  data-testid="heading-1"
>
  Medium Title
</h1>
`;

exports[`Page Components - Snapshots > PageTitle - Snapshots > matches snapshot for default PageTitle 1`] = `
<h1
  class="font-bold text-foreground text-5xl lg:text-6xl"
  data-testid="heading-1"
>
  Default Title
</h1>
`;
